daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -440,7 +446,65 @@ private[lf] case class PartialTransaction(
           nodeSeeds = nodeSeeds :+ (nodeId -> nodeSeed),
         )
       case _ =>
-        noteAbort(Tx.EndExerciseInRootContext)
+        noteAbort(Tx.NonExerciseContext)
+    }
+
+  def abortExercises: PartialTransaction =
+    context.info match {
+      case ec: ExercisesContextInfo =>
+        val exerciseNode = Node.NodeExercises(
+          targetCoid = ec.targetId,
+          templateId = ec.templateId,
+          choiceId = ec.choiceId,
+          optLocation = ec.optLocation,
+          consuming = ec.consuming,
+          actingParties = ec.actingParties,
+          chosenValue = ec.chosenValue,
+          stakeholders = ec.stakeholders,
+          signatories = ec.signatories,
+          choiceObservers = ec.choiceObservers,
+          children = context.children.toImmArray,
+          exerciseResult = None,
+          key = ec.contractKey,
+          byKey = ec.byKey,
+          version = packageToTransactionVersion(ec.templateId.packageId),
+        )
+        val nodeId = ec.nodeId
+        val nodeSeed = context.nextChildSeed
+        copy(
+          context = ec.parent.addChild(nodeId),
+          nodes = nodes.updated(nodeId, exerciseNode),
+          nodeSeeds = nodeSeeds :+ (nodeId -> nodeSeed),
+        )
+      case _ =>
+        noteAbort(Tx.NonExerciseContext)
+    }
+
+  def beginTry: PartialTransaction =
+    copy(context = context.copy(TryContextInfo(context)))
+
+  def endTry: PartialTransaction =
+    context.info match {
+      case info: TryContextInfo =>
+        copy(context = context.copy(info.parent.info))
+      case _ =>
+        noteAbort(Tx.NonCatchContext)
+    }
+
+  def abortTry: PartialTransaction =
+    endTry
+
+  @silent("parameter value exceptionType in method catchAndRollback is never used")
+  @silent("parameter value exception in method catchAndRollback is never used")
+  def catchAndRollback(
```

Comment:
Very small change requested. Can we name this to make it clear that it is one of the 3 possible ways to close a
`beginTry`....
along with `endTry`, `abortTry` 

So perhaps this can be `catchAndRollbackTry`?